statsmodels/tsa/stattools.py

```diff
@@ -1620,3 +1620,279 @@ def _sigma_est_kpss(resids, nobs, lags):
+        elif maxlag:
+            baselags = maxlag
+        else:
+            baselags = int(12. * np.power(nobs / 100., 1 / 4.))
+        trimcnt = int(nobs * trim)
+        start_period = trimcnt
+        end_period = nobs - trimcnt
+        if regression == 'ct':
+            basecols = 5
+        else:
+            basecols = 4
+        # first-diff y and standardize for numerical stability
+        dy = np.diff(x, axis=0)[:, 0]
+        dy /= np.sqrt(dy.T.dot(dy))
+        x = x / np.sqrt(x.T.dot(x))
+        # reserve exog space
+        exog = np.zeros((dy[baselags:].shape[0], basecols + baselags))
+        # normalize constant for stability in long time series
+        c_const = 1 / np.sqrt(nobs)  # Normalize
+        exog[:, 0] = c_const
+        # lagged y and dy
+        exog[:, basecols - 1] = x[baselags:(nobs - 1), 0]
+        exog[:, basecols:] = lagmat(
+            dy, baselags, trim='none')[baselags:exog.shape[0] + baselags]
+        # better time trend: t_const @ t_const = 1 for large nobs
+        t_const = np.arange(1.0, nobs + 2)
+        t_const *= np.sqrt(3) / nobs ** (3 / 2)
+        # iterate through the time periods
+        stats = np.full(end_period + 1, np.inf)
+        for bp in range(start_period + 1, end_period + 1):
+            # update intercept dummy / trend / trend dummy
+            cutoff = (bp - (baselags + 1))
+            if regression != 't':
+                exog[:cutoff, 1] = 0
+                exog[cutoff:, 1] = c_const
+                exog[:, 2] = t_const[(baselags + 2):(nobs + 1)]
+                if regression == 'ct':
+                    exog[:cutoff, 3] = 0
+                    exog[cutoff:, 3] = t_const[1:(nobs - bp + 1)]
+            else:
+                exog[:, 1] = t_const[(baselags + 2):(nobs + 1)]
+                exog[:(cutoff-1), 2] = 0
+                exog[(cutoff-1):, 2] = t_const[0:(nobs - bp + 1)]
+            # check exog rank on first iteration
+            if bp == start_period + 1:
+                o = OLS(dy[baselags:], exog, hasconst=1).fit()
+                if o.df_model < exog.shape[1] - 1:
+                    raise ValueError(
+                        'ZA: auxiliary exog matrix is not full rank.\n \
+                        cols (exc intercept) = {}  rank = {}'.format(
+                            exog.shape[1] - 1, o.df_model))
+                stats[bp] = o.tvalues[basecols - 1]
+            else:
+                stats[bp] = self.__quick_ols(dy[baselags:], exog)[basecols - 1]
+        # return best seen
+        zastat = np.min(stats)
+        bpidx = np.argmin(stats) - 1
+        crit = self.__za_crit(zastat, regression)
+        pval = crit[0]
+        cvdict = crit[1]
+        return zastat, pval, cvdict, baselags, bpidx
+
+    def __call__(self, x, trim=0.15, maxlag=None, regression='c',
+                 autolag='AIC'):
+        return self.run(x, trim=trim, maxlag=maxlag, regression=regression,
+                        autolag=autolag)
+
+za = ur_za()
```

Comment:
Please give this a longer, more informative name.

If the class is t necessary, it’s fine to just implement this as a function.

Reply:
See first comment. I can call the class ZivotAndrewsUnitRoot and the actual call ZivotAndrews? I like the idea of keeping as a class due to the two ZA-specific helpers but can certainly break out if that is the preference. Let me know and I will update the PR as requested.

Reply:
Changed function call name from za() to zivot_andrews (following leinson_durbin naming convention).